src/vos/vos_ts.h

```diff
@@ -730,28 +730,28 @@ vos_ts_set_update(struct vos_ts_set *ts_set, daos_epoch_t read_time)
 /** Save the current state of the set
  *
  * \param[in]	ts_set	The timestamp set
- * \param[out]	statep	Target to save state to
+ * \param[out]	save	Target to save state to
  */
 static inline void
-vos_ts_set_save(struct vos_ts_set *ts_set, int *statep)
+vos_ts_set_save(struct vos_ts_set *ts_set, struct vos_ts_set *save)
 {
 	if (ts_set == NULL)
 		return;
 
-	*statep = ts_set->ts_init_count;
+	*save = *ts_set;
```

Comment:
So you want to save the whole vos_ts_set, right? but vos_ts_set::ts_entries[] is allocated next to the vos_ts_set structure, do we need some special handle for that?

Reply:
I don’t need to save those.   Just the metadata.  So they will be unaffected